ui/common/src/commonMain/kotlin/social/androiddev/common/composables/Spacers.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is part of MastodonX.
+ *
+ * MastodonX is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * MastodonX is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with MastodonX. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package social.androiddev.common.composables
+
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.size
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun RowScope.WeightSpacer(modifier: Modifier = Modifier) {
+    Spacer(modifier = modifier.weight(1f))
+}
+
+@Composable
+fun ColumnScope.WeightSpacer(modifier: Modifier = Modifier) {
+    Spacer(modifier = modifier.weight(1f))
+}
+
+@Composable
+fun XSmallSpacer(modifier: Modifier = Modifier) {
+    Spacer(modifier = modifier.size(2.dp))
+}
+
+@Composable
+fun SmallSpacer(modifier: Modifier = Modifier) {
+    Spacer(modifier = modifier.size(4.dp))
+}
+
+@Composable
+fun DefaultSpacer(modifier: Modifier = Modifier) {
```

Comment:
I have a feeling every time I see DefaultSpacer used I'll need to look it up to see what the actual spacing is. Most likely if we are going to have our own design system we will use common spacings. Maybe we could rename these Spacers to something more descriptive.

Just some ideas:

`ElementSpacing()` // spacing between elements vertically or horizontally like an Avatar and the title after it
`ContainerSpacing()` // standard layout/parent/screen spacing 
...

Reply:
I feel like this collection of composables could grow in size with time, so maybe we could have it scoped into `Spacing` object, so that we have one go-to namespace?

Something like

```kotlin
object Spacing {
  @Composable
  fun Default() {
    Spacer(modifier = Modifier.size(2.dp))
  }

  @Composable
  fun XSmall() {
    Spacer(modifier = Modifier.size(1.dp))
  }
}
```

and usage:

```kotlin
@Composable
fun MyUi() {
  Spacing.Default()

  Spacing.XSmall()
}
```